src/stitcher/image_loader.rs

```diff
@@ -41,15 +41,15 @@ impl From<ImageError> for ImageLoaderError {
     }
 }
 
-impl ImageLoaderError {
-    // Convert a [io::Error] into a [ImageLoaderError].
-    fn from_io_error(err: std::io::Error, path: PathBuf) -> ImageLoaderError {
-        use std::io::ErrorKind as Kind;
-        match err.kind() {
-            Kind::NotFound => ImageLoaderError::NotFound(path),
-            Kind::PermissionDenied => ImageLoaderError::PermissionDenied(path),
+impl From<io::Error> for ImageLoaderError {
```

Comment:
Added a `From<io::Error>` implementation for simpler error handling, replacing the prior `from_io_error` function. You can just use `?` to bubble up IO errors now.